addon/classes/ChangesetArrayProxyHandler.ts

```diff
@@ -0,0 +1,298 @@
+import ArrayProxy from '@ember/array/proxy';
+import { action } from '@ember/object';
+import { A } from '@ember/array';
+import { TrackedMap } from 'tracked-maps-and-sets';
+import IChangesetProxyHandler from './IChangesetProxyHandler';
+import ProxyOptions from './ProxyOptions';
+import { ProxyArrayValueKey } from 'ember-changeset/utils/symbols';
+
+type Input = any[] | ArrayProxy<any>;
+
+export default class ChangesetArrayProxyHandler<T extends Input> implements IChangesetProxyHandler<T> {
+  constructor(source: T, _options?: ProxyOptions) {
+    if (source?.constructor?.name === 'ArrayProxy') {
+      this.__sourceProxy = source as ArrayProxy<any>;
+      this.__source = (source as ArrayProxy<any>).toArray();
+    } else {
+      // wrap the array so it's tracked
+      this.__source = A(source as []);
+    }
+  }
+
+  public get(_target: T, key: string): any {
+    // extra stuff
+    switch (key) {
```

Comment:
you can shorten this by doing `if (typeof this.readArray[key] === 'function') return this.readArray[key];`, etc

Reply:
If you notice I have two different ways of doing it
readArray[key] returns the proxy or the source if there's no changes
writeArray[key] creates a proxy if it doesn't exist and returns it

I thought about using a map to choose which one to return. Otherwise `readKeys.includes[key]` etc

Thoughts?

Reply:
`typeof this.readArray[key] === 'function'` doesn't work - typescript expects the index to be a number. Is there a way to iterate methods on an array without [key]?

Reply:
I aw not sure